snake/snake_game_khr.py

```diff
@@ -0,0 +1,174 @@
+            if i == 0:
+                row = [self.top_boarders] * self.width
+                new_board.append(row)
+            elif i in {self.height - 1}:
+                row = [self.bottom_boarders] * self.width
+                new_board.append(row)
+            else:
+                row = []
+                for j in range(self.width):
+                    if j in {0, self.width - 1}:
+                        row.append(self.vert_boarders)
+                    else:
+                        row.append(" ")
+                new_board.append(row)
+        return new_board
+
+    def show(self):
+        for row in self.board:
+            print(' '.join(row))
+
+
+class Snake:
+    def __init__(self, symbol="o", position=(1, 1)):
+        self.symbol = symbol
+        self.body = [position]
+
+    def eat(self, position: tuple):
+        self.body.append(position)
+
+    def move(self, position: tuple):
+        self.body.append(position)
+        self.body.pop(0)
+
+    def choices(self):
+        next_positions = [
+            (1, 0),
+            (0, 1),
+            (-1, 0),
+            (0, -1)
+        ]
+        result = []
+        for next_position in next_positions:
+            new_positions = self.body[-1][0] + next_position[0], self.body[-1][1] + next_position[1]
+            for new_position in new_positions:
+                if new_position in self.body:
+                    continue
+            result.append(new_positions)
+        return result
+
+
+class Apple:
+    def __init__(self, symbol='$', position=(2, 2)):
+        self.position = position
+        self.symbol = symbol
+
+
+class GameOverError(Exception):
+    pass
+
+
+class Game:
+    def __init__(self, width, height):
+        self.symbol = "o"
+        self.width = width
+        self.height = height
+        self.board = Board(self.width, self.height)
+        self.snake = Snake()
+        self.apple = Apple()
+
+    def init_apple(self):
+        last_pos = self.apple.position
+        self.board = Board(self.width, self.height)
+        positions = []
+        for x in range(self.height):
+            for y in range(self.width):
+                z = (item for item in self.snake.body)
+                if (x, y) != last_pos and (x, y) not in z:
+                    if self.board.board[x][y] == ' ':
+                        positions.append((x, y))
+
+        if positions:
+            position = random.choices(positions)[0]
+            self.apple = Apple(position=position, symbol="$")
+            return self.apple.position
+        else:
+            raise GameOverError('No places for apple!')
+
+    def play(self):
+        game_cycles = 100
+        try:
+            self.render()
+            while game_cycles > 0:
+                i, j = self.snake.body[-1]
+                possible_moves = self.snake.choices()
+                next_moves = []
+                for move in possible_moves:
+                    if move in self.snake.body:
```

Comment:
I thought `possible_moves = self.snake.choices()` wont include moves which are in self.snake.body.

So this line is redundant to me

Reply:
@mykytapavlov 
So, for this one, I'll leave it here and change self.snake.choices() method itself, by making it return only next possible moves for snake (with no condition of self.snake.body intersection). It seems to be more readable in this case.

Reply:
don't agree, but as you wish
In my opinion snake class instance can account it's body in choices